src/js/components/icons/IconUpgradeBlock.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+class IconUpgradeBlock extends React.Component {
+  render() {
+    let iconContent = (<rect width="10" height="10" />);
+
+    if (this.props.hasDecisionPoint) {
```

Comment:
Could we move the logic to user of the icons and have two different icons for these, since they are different icons alltogether?